### PR TITLE
Add General Info step

### LINKED ISSUE #19: (SP: 2) Implement General Info step
**User Stories #77, #74**
## Description
Implement the general info step to used it in both student and tutor cause they are the same according to the mockup.

## Design 

Image

## Acceptance Criteria
- [ ] First name and Last name of a user should be pulled from first registration form.
- [ ] Country and city have to be pulled from server on input focus
- [ ] Additional info length can't be more than 70 symbols

### PR DESCRIPTION
**Pull request by petro97**
Screenshot 2024-10-24 at 11 59 29

### CODE REVIEW

File: src/assets/img/student-home/general-tab-img.svg
Comment (k7bart) [inline src/assets/img/student-home/general-tab-img.svg:1]: We already have this image in '~/assets/img/tutor-home-page/become-tutor/general-info.svg'

**Review by k7bart — APPROVED** [review]

File: src/assets/img/student-home/general-tab-img.svg
Comment (petro97) [reply src/assets/img/student-home/general-tab-img.svg:1]: Thanks! I'll remove it here.

File: src/containers/tutor-home-page/general-info-step/GeneralInfoStep.jsx [hunk 1]
```diff
@@ -1,21 +1,115 @@
...
+              onChange={handleInputChange('lastName')}
+              placeholder={t('common.labels.lastName')}
+              required
+              sx={{ mb: '5px' }}
+              type='text'
+              value={data.lastName}
+            />
+            <AppAutoComplete
+              sx={{ mb: '30px' }}
+              textFieldProps={{
```
Comment (geersann) [inline src/containers/tutor-home-page/general-info-step/GeneralInfoStep.jsx:83]: I think you should add `onChange` and probably `onBlur` for textFieldProps

File: src/containers/tutor-home-page/general-info-step/GeneralInfoStep.jsx [hunk 2]
```diff
@@ -1,21 +1,115 @@
...
+              }}
+              value={data.country}
+            />
+            <AppAutoComplete
+              disabled={!data.country}
+              filterOptions={filterOptions}
+              options={[]}
+              sx={{ mb: '30px' }}
+              textFieldProps={{
+                label: t('common.labels.city')
```
Comment (geersann) [inline src/containers/tutor-home-page/general-info-step/GeneralInfoStep.jsx:94]: I think you should add `onChange` and probably `onBlur` for textFieldProps

**Review by geersann — APPROVED** [review]

File: src/containers/tutor-home-page/general-info-step/GeneralInfoStep.jsx (hunk 1 above)
Comment (petro97) [reply src/containers/tutor-home-page/general-info-step/GeneralInfoStep.jsx:83]: We should get Countries and cities from the back end. There are no endpoints for that. So onChange will be added in the scope of this ticket.https://github.com/WebUI-Project-Based-10-nr/Client/issues/10

File: src/containers/tutor-home-page/general-info-step/GeneralInfoStep.jsx (hunk 2 above)
Comment (petro97) [reply src/containers/tutor-home-page/general-info-step/GeneralInfoStep.jsx:94]: @geersann We should get Countries and cities from the back end. There are no endpoints for that. So onChange will be added in the scope of this ticket.https://github.com/WebUI-Project-Based-10-nr/Client/issues/10

**Review by hanus-kostiantyn — APPROVED** [review]

**Review by nrevutska — APPROVED** [review]

**Pull request merged**